File: Java/SpaceVIL/spacevil/src/main/java/com/spvessel/spacevil/Decorations/ItemState.java
package com.spvessel.spacevil.Decorations;

import java.awt.*;

/**
 * A class that describes state of the item and its changes according
 * to state value.
 */
public class ItemState {
    public Boolean value = true;

    /**
     * Getting backgroud color of the item of current state as java.awt.Color.
     */
    public Color background;

    /**
     * Getting border of the item of current state as com.spvessel.spacevil.Decorations.Border.
     */
    public Border border = new Border();
    
    /**
     * Getting shape of the item of current state as com.spvessel.spacevil.Decorations.Figure.
     */
    public Figure shape = null;

    /**
     * Constructs an empty ItemState
     */
    public ItemState() {

    }

    /**
     * Constructs an ItemState with the specified background color.
     * @param background A color of item as java.awt.Color.
     */
    public ItemState(Color background) {
        this.background = background;
    }
}

File: SpaceVIL/Decorations/ItemState.cs
using System.Drawing;

namespace SpaceVIL.Decorations
{
    /// <summary>
    /// A class that describes state of the item and its changes according to state value.
    /// </summary>
    public class ItemState
    {
        public bool Value = true;

        /// <summary>
        /// Getting backgroud color of the item of current state as System.Drawing.Color.
        /// </summary>
        public Color Background;

        /// <summary>
        /// Getting border of the item of current state as SpaceVIL.Decorations.Border.
        /// </summary>
        public Border Border = new Border();

        /// <summary>
        /// Getting shape of the item of current state as SpaceVIL.Decorations.Figure.
        /// </summary>
        public Figure Shape = null;

        /// <summary>
        /// Constructs an empty ItemState
        /// </summary>
        public ItemState() { }

        /// <summary>
        /// Constructs an ItemState with the specified background color.
        /// </summary>
        /// <param name="background">A color of item as System.Drawing.Color.</param>
        public ItemState(Color background)
        {
            this.Background = background;
        }
    }
}
